File: packages/stronghold_core/lib/models/responses/recommendation_response.dart
/// Supplement recommendation response
class RecommendationResponse {
  final int id;
  final String name;
  final double price;
  final String? description;
  final String? imageUrl;
  final String categoryName;
  final String supplierName;
  final double averageRating;
  final int reviewCount;
  final String recommendationReason;

  const RecommendationResponse({
    required this.id,
    required this.name,
    required this.price,
    this.description,
    this.imageUrl,
    required this.categoryName,
    required this.supplierName,
    required this.averageRating,
    required this.reviewCount,
    required this.recommendationReason,
  });

  factory RecommendationResponse.fromJson(Map<String, dynamic> json) {
    return RecommendationResponse(
      id: json['id'] as int,
      name: json['name'] as String,
      price: (json['price'] as num).toDouble(),
      description: json['description'] as String?,
      imageUrl: json['imageUrl'] as String?,
      categoryName: json['categoryName'] as String,
      supplierName: json['supplierName'] as String,
      averageRating: (json['averageRating'] as num).toDouble(),
      reviewCount: json['reviewCount'] as int,
      recommendationReason: json['recommendationReason'] as String,
    );
  }
}

File: Stronghold/Stronghold.Application/Features/Recommendations/DTOs/RecommendationResponse.cs
namespace Stronghold.Application.Features.Recommendations.DTOs
{
    public class RecommendationResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
        public string CategoryName { get; set; } = string.Empty;
        public string SupplierName { get; set; } = string.Empty;
        public double AverageRating { get; set; }
        public int ReviewCount { get; set; }
        public string RecommendationReason { get; set; } = string.Empty;
    }
}
